ipld/amt/src/amt.rs

```diff
@@ -103,15 +87,60 @@ where
     pub fn count(&self) -> u64 {
         self.root.count
     }
+}
 
+impl<V, BS, Ver> AmtImpl<V, BS, Ver>
+where
+    Ver: AmtVersion,
+    BS: Blockstore,
+    V: Serialize,
+{
     /// Generates an AMT with block store and array of cbor marshallable objects and returns Cid
+    ///
+    /// This can be called with an iterator of _references_ to values to avoid copying.
     pub fn new_from_iter(block_store: BS, vals: impl IntoIterator<Item = V>) -> Result<Cid, Error> {
```

Comment:
Need the bitwidth here; that's another reason I couldn't use this method. Because event HAMTs are not mutable, I think it makes sense to optimise for lower depth, but curious about your opinion. I think a bitwidth of 5 or 8 makes the most sense here?

Reply:
Depends on the max serialized size of an event, really.